src/plugins/intel_gpu/include/intel_gpu/primitives/arg_max_min.hpp

```diff
@@ -41,8 +41,10 @@ struct arg_max_min : public primitive_base<arg_max_min> {
                 ov::op::TopKSortType sort = ov::op::TopKSortType::SORT_VALUES,
                 bool values_first = false,
                 const padding& output_padding = padding(),
-                data_types output_data_type = data_types::f32)
-        : primitive_base(id, {input}, output_padding, optional_data_type {output_data_type}),
+                data_types output_data_type = data_types::f32,
+                const std::vector<input_info>& inputs = {},
```

Comment:
The reason why we added additioanl input arg is not to replace other primitives, so that this PR is applied as a intermediate version. 

However we may be able to do like this 
1) Add additional permute primitive API where the input is 
2) Have two versions of primitive_base API, 1) original one 2) accepting input as vector of input_info
3) If the original primitive_base API is used, set the input_info::idx as 0

Reply:
How the other primitives are affected by arg_max_min c-tor? If you replace primitive_ids vector with input_info, and then just extract ids to initialize primitive base field, you won't even need to add new c-tor for primitive_base (though adding it would be better)

Reply:
@vladimir-paramuzov  Ah what I meant was the API for primitive_base. We can either change or add additional APIs w/ input_info instead of inputs for the used primitives here. 

In my thought,  "adding a new API" is not preferrable, so we'd better change the existing APIs. To do so we' ll need to update CreateOps too. We are going to update CreateOps in the next PR, so I think they can be modified at that PR. 

How do you think doing it in the next PR?

Reply:
OK